planb/PBManifest.m

```diff
@@ -159,11 +166,15 @@ - (BOOL)validateTrack:(id)object {
     return NO;
   }
   NSDictionary *item = object;
-  for (NSString *key in @[kFilenameKey, kSHA256Key]) {
-    if (![item[key] isKindOfClass:[NSString class]]) {
-      PBLog(@"Invalid manifest -- track value for %@ key is not a string: %@", key, item[key]);
-      return NO;
-    }
+  if (![item[kFilenameKey] isKindOfClass:[NSString class]]) {
+    PBLog(@"Invalid manifest -- track value for %@ key is not a string: %@", kFilenameKey,
+          item[kFilenameKey]);
+    return NO;
+  }
+  if (item[kSHA256Key] && ![item[kSHA256Key] isKindOfClass:[NSString class]]) {
```

Comment:
same

Reply:
The sha256 key is also optional.  I thought about maybe making a more generic validator object for all of these that would include the key name, it's desired class, and whether or not it was optional, then looping over everything, but it seemed like a lot of overhead to set up.  I could maybe make it into a function call like ValidateKey(dict, name, class, optional).  But then the error logs would lose some of their specificity.  So I don't know.

Reply:
This is fine, maybe just add a comment. The extra check is for logic not safety.

Reply:
Done.